src/main/java/eu/europeana/metis/sandbox/service/util/XsltUrlUpdateServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package eu.europeana.metis.sandbox.service.util;
+
+import eu.europeana.metis.sandbox.entity.TransformXsltEntity;
+import eu.europeana.metis.sandbox.repository.TransformXsltRepository;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.charset.StandardCharsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class XsltUrlUpdateServiceImpl implements XsltUrlUpdateService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XsltUrlUpdateServiceImpl.class);
+  private static final HttpClient httpClient = HttpClient.newBuilder().build();
+  private final TransformXsltRepository transformXsltRepository;
+
+  public XsltUrlUpdateServiceImpl(TransformXsltRepository transformXsltRepository) {
+    this.transformXsltRepository = transformXsltRepository;
+  }
+
+  @Override
+  @Transactional
+  public void updateXslt(String defaultXsltUrl) {
+
+    HttpRequest httpRequest = HttpRequest.newBuilder()
+        .GET()
+        .uri(URI.create(defaultXsltUrl))
+        .build();
+
+    try (final InputStream xsltStream = httpClient.send(httpRequest, BodyHandlers.ofInputStream())
+        .body()) {
+      String transformXslt = new String(xsltStream.readAllBytes(),
+          StandardCharsets.UTF_8);
+      var entity = transformXsltRepository.findByTransformXslt(transformXslt);
+
+
+      if (entity != null && entity.isPresent()) {
+        if (!(transformXslt.equals(entity.get().getTransformXslt()))){
```

Comment:
This is still incorrect.
That is because the fetching from the database of transformXsltRepository.findByTransformXslt(transformXslt); above!
Imagine this scenario:
First time we are getting from the URL String "A" and since we don't find it we are storing String "A".
Second time we are getting from the URL String "B" and we  are searching with that and since we don't find it we store it.
So essentially we are creating new ones all the time.

The idea is to replace the existent one.
What you are missing here is searching for the right thing, you'd need to search not based on the String received but rather at the first(there should be only one) item from the database and replace that if it's different.